### DISCUSSION TITLE
How to deploy inside Portainer or in a containerized environment with SSL

### DISCUSSION THREAD

**Discussion by antonymarion** [OvenMediaLabs/OvenMediaEngine#1666] [Help] [answered]
Hello OME team,

We use your product and love it!

We have one question regarding the Server.xml file inside the conf. directory.

We have put the certs files 
![image](https://github.com/user-attachments/assets/c247b9e5-3c11-4821-9477-d8d6eda69f04)

and kept the Server.xml like below:
![image](https://github.com/user-attachments/assets/f5972cb1-f369-4ecc-9608-35b9410d3b38)

Nevertheless, we cannot use the API when curling to https://{our_domain}/v1/stats/current/vhosts/test
And we are not able to retrieve the stream with OvenPlayer when we put in the url: wss://{our_domain}:3334/app/{stream_key}

By the way, here is the cmd that we use to send the stream:

` ffmpeg -d -stream_loop -1 -re -i myFile.mp4 -c copy -f flv -y rtmp://{our_domain}:1935/app/{stream_key}`
and we verified that this command is working well with a local and a public ip version of OME on http (no secured) and using ws (not secured)

**Comment by getroot** [comment]
There will be a hint in the log. If you upload the log file and the entire Server.xml, I will check it.

**Reply by antonymarion** [reply]
Here are the logs starting from the 18/07/24
[_ome_logs (4).txt](https://github.com/user-attachments/files/16312093/_ome_logs.4.txt)

**Reply by antonymarion** [reply]
and the Server.xml 
[Server.txt](https://github.com/user-attachments/files/16312129/Server.txt)

**Reply by naanlizard** [reply]
This doesn't appear to be the full log, you need to include the beginning, where it would load the certs.

The ffmpeg command doesn't interact with the ssl stuff at all, it connects to an entirely different port. Presumably you could have http and ws working fine if the certs aren't in the correct place in the container

I'd recommend against relative paths for the cert, but I think they do work in theory. Are the certs there when the container starts?

**Reply by BhushanKakde** [reply]
bro can you tell me what type of work this is

**Reply by BhushanKakde** [reply]
iam new

**Reply by antonymarion** [reply]
> This doesn't appear to be the full log, you need to include the beginning, where it would load the certs.
> 
> The ffmpeg command doesn't interact with the ssl stuff at all, it connects to an entirely different port. Presumably you could have http and ws working fine if the certs aren't in the correct place in the container
> 
> I'd recommend against relative paths for the cert, but I think they do work in theory. Are the certs there when the container starts?

Sure I will try to track the cert part in the full logs and send you the log.

Will send you this tomorrow morning.

To answer your question : the certs were not there initially, but placed after by our DevOps.

And I believe this is the root cause of the issue. Will check that tomorrow, thks !

If it is the cause, is there a way to restart the ome server inside the container without restarting the whole container ? Or a way maybe to dynamically update the cert configuration of the ome when it is already running?

**Reply by antonymarion** [reply]
Please find attached the full logs 
[_ome_logs (6).txt](https://github.com/user-attachments/files/16323757/_ome_logs.6.txt)

**Reply by antonymarion** [reply]
certs loading are ok from what I see from the logs here:
![image](https://github.com/user-attachments/assets/0b891730-67d9-42fb-98ab-798aa3ba4962)

**Reply by antonymarion** [reply]
When I start streaming it seems ok from the log I get from the docker container:
![image](https://github.com/user-attachments/assets/3288ee4a-c860-4027-a080-1b9249a42d6a)

But I am unable then to target the stream using wss:

![image](https://github.com/user-attachments/assets/f2509370-1898-480b-aeaf-41fe7c637eda)

**Comment by naanlizard** [comment]
My next guess is the docker container's ports are not accessible for
whatever reason. Can you post your docker run command or compose file

**Reply by antonymarion** [reply]
Sure, please find below the whole conf. from portainer:

![image](https://github.com/user-attachments/assets/23866b2c-c4c6-4205-b32f-98cf1112747c)
![image](https://github.com/user-attachments/assets/ce7be040-4307-4391-8d1f-d091106f475d)

**Reply by antonymarion** [reply]
Ohhh it seems that 3334 port (SSL port for ws) is not open right ? so that's the reason I believe

**Reply by antonymarion** [reply]
Well, I added 3334 port binding between host and container (inside portainer conf.) and now I am getting the following console log on our https player:

![image](https://github.com/user-attachments/assets/a2a40ace-82f7-414f-839c-4797abf6d496)

**Comment by naanlizard** [comment]
What does the OME log show?

**Reply by antonymarion** [reply]
![image](https://github.com/user-attachments/assets/9548fc1b-0137-4937-b456-e2da5ae3e913)

**Reply by antonymarion** [reply]
Ome logs look good

**Reply by getroot** [reply]
Set the OME_HOST_IP environment variable to the IP address that the player can access.

Your log has the following warning:
[2024-07-18 09:43:31.186] W [OvenMediaEngine:1] ICE | ice_port_manager.cpp:298 | Invalid address: Your.HOST.IP.Address, port: 10000 - [SocketAddress] Name or service is unknown: "Your.HOST.IP.Address"

Also, 0.16.7 does not work properly, although it is now removed. Check if it is 0.16.8.

**Reply by naanlizard** [reply]
If OME_HOST_IP isn't set, what happens? I don't think we have that set, and are looking to upgrade to 16.8 soon

**Reply by getroot** [reply, marked as answer]
If OME_HOST_IP is not set at all, it will work with the default setting, but in the above case, it will act as if Your.HOST.IP.Address is a domain, which may cause an error. In the case of default, OME will get all IPs from the NIC, use STUN to get IPs, and use all of them as candidates. In this case, the initial connection may be slightly slower because the Player will try all the candidates provided. This is also necessary because in cases like AWS, the public IP is not known inside the instance and Stun used by OME may not work. That's why we recommend OME_HOST_IP.

**Reply by antonymarion** [reply]
Indeed env var was missing, thks for your swift support guys!
